Src/GUI/Widgets/OutputFilesWidget.py

```diff
@@ -56,17 +78,17 @@ def get_widget_value(self):
                                     
         return (filename, formats)
     
-    def set_data(self, datakey):
-
-        basename = "output"
+    def set_data(self, datakey, analysis):
                 
         if datakey is None:
-            basename = "output"
+            basename = "output_%s" % analysis
             trajectoryDir = os.getcwd()
         else:
-            basename = "output_%s" % os.path.splitext(os.path.basename(datakey))[0]
+            basename = "%s_%s" % (os.path.splitext(os.path.basename(datakey))[0], analysis)
             trajectoryDir = os.path.dirname(datakey)
-            
-        self._filename.SetValue(os.path.join(trajectoryDir,basename))
+
+        path = OutputFilesWidget._get_unique_filename(trajectoryDir,basename)

```

Comment:
Similarly, is there a reason to reference this class via `OutputFilesWidget` rather than `self` here?

Reply:
You can use both. It is a question of styling. `_get_unique_name` being a staticmethod, I found more descriptive to use it through the class object.